src/scripts/main.js

```diff
@@ -1 +1,52 @@
 'use strict';
+
+const firstPromise = new Promise(function (resolve, reject) {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+
+    setTimeout(() => {
+      reject(new Error(`First promise was rejected`));
+    }, 3000);
```

Comment:
The timeout logic here creates a race condition. The promise could be rejected after 3 seconds even if the left mouse button is clicked. Consider using a mechanism to clear the timeout if the promise is resolved.